Fuzz Testing & Nightly Cargo udeps are failing due to prost-derive

As mentioned, Fuzz Testing & Nightly Cargo udeps are failing due to prost-derive.

```
   Compiling prost-derive v0.6.1
error[E0034]: multiple applicable items in scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/prost-derive-0.6.1/src/lib.rs:111:14
    |
111 |             .intersperse(quote!(|));
    |              ^^^^^^^^^^^ multiple `intersperse` found
    |
    = note: candidate #1 is defined in an impl of the trait `Iterator` for the type `Map<I, F>`
    = note: candidate #2 is defined in an impl of the trait `Itertools` for the type `T`
help: disambiguate the associated function for candidate #1
    |
107 ~         let tags = Iterator::intersperse(field
108 +             .tags()
109 +             .into_iter()
110 +             .map(|tag| quote!(#tag)), {
111 +         let mut _s = $crate::__private::TokenStream::new();
112 +         $crate::quote_each_token!(_s $($tt)*);
  ...
help: disambiguate the associated function for candidate #2
    |
107 ~         let tags = Itertools::intersperse(field
108 +             .tags()
109 +             .into_iter()
110 +             .map(|tag| quote!(#tag)), {
111 +         let mut _s = $crate::__private::TokenStream::new();
112 +         $crate::quote_each_token!(_s $($tt)*);
```

--- a/.cargo/audit.toml
+++ b/.cargo/audit.toml
@@ -1,6 +1,5 @@
 [advisories]
-ignore = ["RUSTSEC-2021-0073", # We do not use `prost_types::Timestamp` anywhere in our code.
-          "RUSTSEC-2020-0036"] # We don't have control over the exact dependencies of `protoc-grpcio`; See https://github.com/mtp401/protoc-grpcio/issues/36
+ignore = []
 informational_warnings = ["unmaintained"] # warn for categories of informational advisories
 severity_threshold = "low" # CVSS severity ("none", "low", "medium", "high", "critical")
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -96,20 +96,19 @@ jobs:
       # When running the container built on the CI
       # run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec -t parsec-service-test-all /tmp/parsec/ci.sh cryptoauthlib --no-stress-test
 
-  # Disabled due to the issue discussed in https://github.com/parallaxsecond/parsec/issues/514
-  # fuzz-test-checker:
-  #   name: Check that the fuzz testing framework is still working
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #     # Use the following step when updating the `parsec-service-test-all` image
-  #     # - name: Build the container
-  #     #   run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
-  #     - name: Run the fuzz test script
-  #       # Not running stress tests because rust-cryptoauthlib test-interface does not support required calls
-  #       run: ./fuzz.sh test
-  #     # When running the container built on the CI
-  #     # run: CONTAINER_TAG=parsec-service-test-all ./fuzz.sh test
+  fuzz-test-checker:
+    name: Check that the fuzz testing framework is still working
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # Use the following step when updating the `parsec-service-test-all` image
+      # - name: Build the container
+      #   run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
+      - name: Run the fuzz test script
+        # Not running stress tests because rust-cryptoauthlib test-interface does not support required calls
+        run: ./fuzz.sh test
+      # When running the container built on the CI
+      # run: CONTAINER_TAG=parsec-service-test-all ./fuzz.sh test
 
   cross-compilation:
     # Currently only the Mbed Crypto, PKCS 11, and TPM providers are tested as the other ones need to cross-compile other libraries.

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -12,22 +12,21 @@ on:
         default: ""
 
 jobs:
-  # Disabled due to the issue discussed in https://github.com/parallaxsecond/parsec/issues/514
-  # dependencies:
-  #   name: Check for unused dependencies
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #       with:
-  #         ref: "${{ github.event.inputs.rev }}"
-  #     - name: Install latest Rust
-  #       uses: actions-rs/toolchain@v1
-  #       with:
-  #         toolchain: nightly
-  #     - name: Install cargo udeps
-  #       run: cargo install cargo-udeps --locked
-  #     - name: Execute cargo udeps
-  #       run: cargo +nightly udeps
+  dependencies:
+    name: Check for unused dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: "${{ github.event.inputs.rev }}"
+      - name: Install latest Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - name: Install cargo udeps
+        run: cargo install cargo-udeps --locked
+      - name: Execute cargo udeps
+        run: cargo +nightly udeps
 
   audit:
     name: Check for crates with security vulnerabilities

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ name = "parsec"
 path = "src/bin/main.rs"
 
 [dependencies]
-parsec-interface = "0.25.0"
+parsec-interface = { git = "https://github.com/parallaxsecond/parsec-interface-rs", rev = "e5b7f17ac192722befd37698a07a7eb4ffabd2a4" }
 rand = { version = "0.8.3", features = ["small_rng"], optional = true }
 base64 = "0.13.0"
 uuid = "0.8.2"
@@ -42,8 +42,8 @@ users = "0.11.0"
 libc = "0.2.86"
 anyhow = "1.0.38"
 rust-cryptoauthlib = { version = "0.4.0", optional = true }
-spiffe = { version = "0.1.1", optional = true }
-prost = { version = "0.7.0", optional = true }
+spiffe = { version = "0.2.0", optional = true }
+prost = { version = "0.8.0", optional = true }
 
 [dev-dependencies]
 rand = { version = "0.8.3", features = ["small_rng"] }
@@ -51,7 +51,7 @@ rust-cryptoauthlib = { version = "0.4.0", features=["software-backend"]}
 
 [build-dependencies]
 bindgen = { version = "0.57.0", optional = true }
-prost-build = { version = "0.7.0", optional = true }
+prost-build = { version = "0.8.0", optional = true }
 
 [package.metadata.docs.rs]
 features = ["pkcs11-provider", "tpm-provider", "mbed-crypto-provider", "cryptoauthlib-provider", "direct-authenticator"]

--- a/ci.sh
+++ b/ci.sh
@@ -249,12 +249,12 @@ if [ "$PROVIDER_NAME" = "cargo-check" ]; then
     # - RHEL-8
     # - openSUSE Tumbleweed
     # - openSUSE Leap 15.3
-    # The oldest is currently in openSUSE Leap 15.3 and is 1.43.0.
+    # The oldest is currently in openSUSE Leap 15.3 and is 1.53.0.
     rustup update
 
-    rustup toolchain install 1.43.0
-    # The "jwt-svid-authenticator" can not be compiled on 1.43.0
-    RUST_BACKTRACE=1 cargo +1.43.0 check --release --features=all-providers,direct-authenticator,unix-peer-credentials-authenticator
+    rustup toolchain install 1.53.0
+    # The "jwt-svid-authenticator" can not be compiled on 1.53.0
+    RUST_BACKTRACE=1 cargo +1.53.0 check --release --features=all-providers,direct-authenticator,unix-peer-credentials-authenticator
 
     # Latest stable
     rustup toolchain install stable

--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -20,7 +20,7 @@ lazy_static = "1.4.0"
 arbitrary = { version = "0.4.0", features = ["derive"] }
 
 [build-dependencies]
-parsec-client = "0.13.0"
+parsec-client = { git = "https://github.com/parallaxsecond/parsec-client-rust", rev = "4579ec1df97531f31d4f55eab4e9b55f2b26399b" }
 
 # Prevent this from interfering with workspaces
 [workspace]

--- a/src/authenticators/jwt_svid_authenticator/mod.rs
+++ b/src/authenticators/jwt_svid_authenticator/mod.rs
@@ -63,14 +63,14 @@ impl Authenticate for JwtSvidAuthenticator {
             ResponseStatus::InvalidEncoding
         })?;
 
-        let (spiffe_id, _) = self
+        let jwt_token = self
             .client
             .validate_jwt_token("parsec", &svid)
             .map_err(|e| {
                 error!("The validation of the JWT-SVID failed ({}).", e);
                 ResponseStatus::AuthenticationError
             })?;
-        let app_name = spiffe_id.to_string();
+        let app_name = jwt_token.spiffe_id().to_string();
         let is_admin = self.admins.is_admin(&app_name);
         Ok(Application::new(app_name, is_admin))
     }

